terraform/production/ci_iam.tf

```diff
@@ -0,0 +1,78 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+resource "aws_iam_role" "github_actions_delta_auth_ci" {
+  name               = "github-actions-delta-auth-ci"
+  assume_role_policy = data.aws_iam_policy_document.github_actions_delta_auth_assume_role.json
+}
+
+data "aws_iam_openid_connect_provider" "github" {
+  arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/token.actions.githubusercontent.com"
+}
+
+data "aws_iam_policy_document" "github_actions_delta_auth_assume_role" {
+  statement {
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+
+    principals {
+      type        = "Federated"
+      identifiers = [data.aws_iam_openid_connect_provider.github.arn]
+    }
+
+    condition {
+      test     = "StringEquals"
+      values   = ["sts.amazonaws.com"]
+      variable = "token.actions.githubusercontent.com:aud"
+    }
+
+    condition {
+      test = "StringLike"
+      values = [
+        "repo:communitiesuk/delta-auth-service:*"
```

Comment:
This should be restricted to the "publish" environment

Reply:
changed to restrict to publish